app/services/allocation_copy_data_service.rb

```diff
@@ -0,0 +1,67 @@
+class AllocationCopyDataService
+  include ServicePattern
+
+  def initialize(allocation_cycle_year:)
+    @allocation_cycle_year = allocation_cycle_year
+  end
+
+  def call
+    Allocation.where(recruitment_cycle_id: previous_cycle.id).find_each do |prev_alloc|
+      next if prev_alloc.confirmed_number_of_places.to_i.zero? ||
+        allocation_exists?(prev_alloc)
+
+      provider = find_provider(prev_alloc)
+      accredited_body = find_accredited_body(prev_alloc)
+      create_allocation(prev_alloc, provider, accredited_body)
+    end
```

Comment:
Is there anything useful that we'd like to know at the end of running this task? Maybe how many we copied over vs skipped?